### PR TITLE
RabbitMQ Weak Credentials Plugin

### PR DESCRIPTION
**Pull request by leonardo-doyensec**
The following PR adds the weak credential tester for RabbitMQ Management Portal.

Some notes on the current version of the rabbitmq management portal credential tester.

- does a custom fingerprint in the `canAccept()` method since rabbitmq is not correctly recognized from nmap
- tests the credentials using Basic Authentication
- the management portal needs to be exposed on a different port, because the default one, which is 15672, is not scanned by the default configuration of nmap

### CODE REVIEW

File: google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+import com.google.tsunami.proto.NetworkService;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+import javax.inject.Inject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+/** Credential tester for RabbitMQ Management Portal. */
+public final class RabbitMQCredentialTester extends CredentialTester {
```
Comment (tooryx) [inline google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java:44]: The class is missing implementation for the [`batched`](https://github.com/google/tsunami-security-scanner-plugins/blob/985f273e6779966806d608e866908e79b79c7a51/google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/tester/CredentialTester.java#L39) method:

```
RabbitMQCredentialTester.java:44: error: RabbitMQCredentialTester is not abstract and does not override abstract method batched() in CredentialTester
```

Note: It seems like the doyensec repository is behind. You might need to sync first.

**Review by tooryx — COMMENTED** [review]
Hi @leonardo-doyensec,

Thank you for your contribution! I just did a very quick review (for example, I did not look at the tests yet) on this plugin and you will find my suggestions below.

Feel free to push back if you disagree with anything.

_~tooryx_

**Comment by leonardo-doyensec** [comment]
Hi @tooryx,
I've performed all the requested changes.
Let me know if there is something else wrong.

Thank you for your time.
Leonardo

**Comment by tooryx** [comment]
Thank you for reviewing and taking the time @leonardo-doyensec! But I think you merged the changes from torchserve also in your `rabbitmq` branch. This PR now contains both plugins which will cause issues with both the bots and the overall review process.

Could you split the changes? It also seems that this introduced some CLA problems.

_~tooryx_

**Comment by leonardo-doyensec** [comment]
Hi @tooryx,
I've messed up a bit with the merging, i'm really sorry.
To keep the pull request clean, i will close this one and move everything to a new one that you can find here
https://github.com/google/tsunami-security-scanner-plugins/pull/372

Since i have cherry picked the commits, everything should be fine with your last requirements.
Sorry once again for the mistake, and thank you for your time and patience.

Leonardo

**Pull request closed without merging**